docs/content/guides/dagster/re-execution.mdx

```diff
@@ -0,0 +1,127 @@
+from random import random
+from dagster import ModeDefinition, fs_io_manager, pipeline, solid
+
+
+@solid()
+def unreliable_start(_):
+    return 1
+
+
+@solid()
+def unreliable(_, num):
+    failure_rate = 0.5
+    if random() < failure_rate:
+        raise Exception("blah")
+
+
+@solid()
+def unreliable_end(_, num):
+    return
+
+
+@pipeline(mode_defs=[ModeDefinition(resource_defs={"io_manager": fs_io_manager})])
+def unreliable_pipeline():
+    unreliable_end(unreliable(unreliable_start()))
+```
+
+Although very simple, there are inputs and outputs passed between tasks. With an IO manager, re-execution is able to handle inputs and outputs stored from the initial run.
+
+Upon launching the execution in Dagit, you can find the re-execution option on the top right.
+
+### Re-execution in Dagit
+
+Under the re-execution drop down, you will see multiple options for different types of re-execution. No matter which one you choose, the re-executed run is linked to the original run.
+
+- **All Steps**: Re-execute the pipeline run from scratch. This option is most relevant upon testing pipeline runs end to end, if you'd like to associate different runs together.
+- **Selected Step(s)**: Re-execute the selected step(s). This can be done independent of the step status. Your pipeline is likely large, and you know exactly which steps to execute. This can be done by clicking on the boxes in the gantt chart view.
+- **From Selected**: Re-execute the pipeline downstream from the selected steps. If a particular step fails, this option is most relevant if your intent is to run all downstream steps regardless of status. You're likely just developing a single step, and want to make sure downstream steps work as expected.
+- **From Failure**: Re-execute the pipeline run, skipping steps completed successfully. This is only enabled when the pipeline has failed. You have likely fixed the failed step, and want to run the step and all downstream dependencies again. Dagster will figure out the dependencies for you!
+
+In the above example, re-executing from failure would make sense as the failed task has a 50% chance of succeeding on the next run.
+
+<Image
+alt="Dagit Re-execution"
+src="/images/guides/reexecution/reexecution_options.png"
+width={3808}
+height={2414}
+/>
+
+If the entire pipeline succeeded but the underlying code changed, running specific steps to test the differences would be more relevant.
+
+### Selecting Steps
+
+Within Dagit, a single or multiple steps may be selected simply by clicking them with the mouse. Alternatively, the text-box can be leveraged by typing step names to be re-run, adding a `+` for every subsequent step to be selected. This is demoed in the video below:
+
+<iframe
+  width="560"
+  height="315"
+  src="https://www.youtube.com/embed/LWc8MxSN22c"
+  frameBorder="0"
+  allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
+  allowfullscreen
+></iframe>
```

Comment:
If possible, could you make the selection video based on the above example so it's consistent? And I'm curious if it's possible/easy to make it a gif that we can include natively in the content.

Reply:
1) + 2) Ah! Moving too fast. Removed the storage files and re-added the re-execution.png file.
3) and 4) Makes sense - I'll compress the screenshots and make another video today.

Reply:
2 and 4 addressed: screenshots compressed, new Youtube video uploaded